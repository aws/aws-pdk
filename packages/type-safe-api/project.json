{
  "name": "@aws/type-safe-api",
  "root": "packages/type-safe-api",
  "targets": {
    "build": {
      "inputs": [
        "default",
        "^default",
        "!{projectRoot}/.jsii",
        "!{projectRoot}/lib/**/*",
        "!{projectRoot}/dist/**/*",
        "!{projectRoot}/coverage/**/*",
        "!{projectRoot}/test-reports/**/*",
        "!{projectRoot}/docs/api/**/*"
      ],
      "outputs": [
        "{projectRoot}/.jsii",
        "{projectRoot}/lib",
        "{projectRoot}/dist",
        "{projectRoot}/coverage",
        "{projectRoot}/test-reports",
        "{projectRoot}/docs/api"
      ],
      "dependsOn": [
        "^build"
      ],
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen build",
        "cwd": "packages/type-safe-api"
      }
    },
    "default": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen default",
        "cwd": "packages/type-safe-api"
      }
    },
    "pre-compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen pre-compile",
        "cwd": "packages/type-safe-api"
      }
    },
    "compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen compile",
        "cwd": "packages/type-safe-api"
      }
    },
    "post-compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen post-compile",
        "cwd": "packages/type-safe-api"
      }
    },
    "test": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen test",
        "cwd": "packages/type-safe-api"
      }
    },
    "package": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package",
        "cwd": "packages/type-safe-api"
      }
    },
    "test:watch": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen test:watch",
        "cwd": "packages/type-safe-api"
      }
    },
    "watch": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen watch",
        "cwd": "packages/type-safe-api"
      }
    },
    "eslint": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen eslint",
        "cwd": "packages/type-safe-api"
      }
    },
    "compat": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen compat",
        "cwd": "packages/type-safe-api"
      }
    },
    "package-all": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package-all",
        "cwd": "packages/type-safe-api"
      }
    },
    "package:java": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package:java",
        "cwd": "packages/type-safe-api"
      }
    },
    "package:python": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package:python",
        "cwd": "packages/type-safe-api"
      }
    },
    "eslint-staged": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen eslint-staged",
        "cwd": "packages/type-safe-api"
      }
    },
    "jest": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen jest",
        "cwd": "packages/type-safe-api"
      }
    },
    "docgen": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen docgen",
        "cwd": "packages/type-safe-api"
      }
    }
  },
  "implicitDependencies": [
    "smithy-async-transformer"
  ],
  "//": "~~ Generated by projen. To modify, edit .projenrc.js and run \"pnpm exec projen\"."
}
