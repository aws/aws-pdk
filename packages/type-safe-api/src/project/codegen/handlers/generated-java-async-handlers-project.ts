/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import {
  GeneratedJavaHandlersBaseProject,
  GeneratedJavaHandlersBaseProjectOptions,
} from "./generated-java-handlers-base-project";
import { CodegenOptions, OtherGenerators } from "../components/utils";

export interface GeneratedJavaAsyncHandlersProjectOptions
  extends GeneratedJavaHandlersBaseProjectOptions {}

export class GeneratedJavaAsyncHandlersProject extends GeneratedJavaHandlersBaseProject {
  constructor(options: GeneratedJavaAsyncHandlersProjectOptions) {
    super(options);
  }

  public buildCodegenOptions(): CodegenOptions {
    return {
      specPath: this.options.specPath,
      templateDirs: [OtherGenerators.JAVA_ASYNC_LAMBDA_HANDLERS],
      metadata: {
        srcDir: this.srcDir,
        tstDir: this.tstDir,
        packageName: this.packageName,
        runtimePackageName: this.options.generatedJavaTypes.packageName,
      },
    };
  }
}
