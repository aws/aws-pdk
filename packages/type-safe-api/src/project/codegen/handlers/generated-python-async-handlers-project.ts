/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import {
  GeneratedPythonHandlersBaseProject,
  GeneratedPythonHandlersBaseProjectOptions,
} from "./generated-python-handlers-base-project";
import { CodegenOptions, OtherGenerators } from "../components/utils";

export interface GeneratedPythonAsyncHandlersProjectOptions
  extends GeneratedPythonHandlersBaseProjectOptions {}

export class GeneratedPythonAsyncHandlersProject extends GeneratedPythonHandlersBaseProject {
  constructor(options: GeneratedPythonAsyncHandlersProjectOptions) {
    super(options);
  }

  public buildCodegenOptions(): CodegenOptions {
    return {
      specPath: this.options.specPath,
      templateDirs: [OtherGenerators.PYTHON_ASYNC_LAMBDA_HANDLERS],
      metadata: {
        srcDir: this.moduleName,
        tstDir: this.tstDir,
        runtimeModuleName: this.options.generatedPythonTypes.moduleName,
        moduleName: this.moduleName,
      },
    };
  }
}
