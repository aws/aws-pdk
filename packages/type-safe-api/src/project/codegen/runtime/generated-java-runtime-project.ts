/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import * as path from "path";
import { JavaProject } from "projen/lib/java";
import { Language } from "../../languages";
import { GeneratedJavaProjectOptions } from "../../types";
import { OpenApiGeneratorIgnoreFile } from "../components/open-api-generator-ignore-file";
import { OpenApiToolsJsonFile } from "../components/open-api-tools-json-file";
import {
  buildCleanOpenApiGeneratedCodeCommand,
  buildInvokeOpenApiGeneratorCommand,
} from "../components/utils";

/**
 * Configuration for the generated java runtime project
 */
export interface GeneratedJavaTypesProjectOptions
  extends GeneratedJavaProjectOptions {
  /**
   * The path to the OpenAPI specification, relative to this project's outdir
   */
  readonly specPath: string;
}

const DEPENDENCIES: string[] = [
  // Required for open api generated code
  "io.swagger/swagger-annotations@1.6.8",
  "com.google.code.findbugs/jsr305@3.0.2",
  "com.squareup.okhttp3/okhttp@4.10.0",
  "com.squareup.okhttp3/logging-interceptor@4.10.0",
  "com.google.code.gson/gson@2.9.1",
  "io.gsonfire/gson-fire@1.8.5",
  "org.apache.commons/commons-lang3@3.12.0",
  "jakarta.annotation/jakarta.annotation-api@1.3.5",
  "org.openapitools/jackson-databind-nullable@0.2.4",
  "javax.ws.rs/jsr311-api@1.1.1",
  "javax.ws.rs/javax.ws.rs-api@2.1.1",
  // For handler wrappers
  "com.amazonaws/aws-lambda-java-core@1.2.1",
  "com.amazonaws/aws-lambda-java-events@3.11.0",
  // Lombok is used to add the builder pattern to models for neater construction
  "org.projectlombok/lombok@1.18.24",
];

const TEST_DEPENDENCIES: string[] = [
  "org.junit.jupiter/junit-jupiter-api@5.9.1",
  "org.mockito/mockito-core@3.12.4",
];

/**
 * Java project containing types generated using OpenAPI Generator CLI
 */
export class GeneratedJavaRuntimeProject extends JavaProject {
  /**
   * Patterns that are excluded from code generation
   */
  public static openApiIgnorePatterns: string[] = [
    "pom.xml",
    "gradle",
    "gradle/**/*",
    "gradlew",
    "gradle.properties",
    "gradlew.bat",
    "build.gradle",
    "settings.gradle",
    "build.sbt",
    ".travis.yml",
    "git_push.sh",
    "src/test",
    "src/test/**/*",
    "src/main/AndroidManifest.xml",
  ];

  /**
   * The package name, for use in imports
   */
  public readonly packageName: string;

  /**
   * Path to the openapi specification
   * @private
   */
  private readonly specPath: string;

  constructor(options: GeneratedJavaTypesProjectOptions) {
    super({
      sample: false,
      junit: false,
      ...options,
    });
    this.specPath = options.specPath;

    // Ignore files that we will control via projen
    const ignoreFile = new OpenApiGeneratorIgnoreFile(this);
    ignoreFile.addPatterns(
      ...GeneratedJavaRuntimeProject.openApiIgnorePatterns
    );

    // Add OpenAPI Generator cli configuration
    OpenApiToolsJsonFile.ensure(this).addOpenApiGeneratorCliConfig(
      options.openApiGeneratorCliConfig
    );

    // Add dependencies
    DEPENDENCIES.forEach((dep) => this.addDependency(dep));
    TEST_DEPENDENCIES.forEach((dep) => this.addTestDependency(dep));

    this.packageName = `${this.pom.groupId}.${this.name}.runtime`;

    // Generate the java code
    const generateCodeCommand = this.buildGenerateCommand();
    const cleanCommand = buildCleanOpenApiGeneratedCodeCommand(this.outdir);

    const generateTask = this.addTask("generate");
    generateTask.exec(cleanCommand.command, {
      cwd: path.relative(this.outdir, cleanCommand.workingDir),
    });
    generateTask.exec(generateCodeCommand.command, {
      cwd: path.relative(this.outdir, generateCodeCommand.workingDir),
    });

    this.preCompileTask.spawn(generateTask);

    // Ignore all the generated code
    this.gitignore.addPatterns(
      "src",
      "docs",
      "api",
      "README.md",
      ".openapi-generator"
    );
  }

  public buildGenerateCommand = () => {
    return buildInvokeOpenApiGeneratorCommand({
      generator: "java",
      specPath: this.specPath,
      outputPath: this.outdir,
      generatorDirectory: Language.JAVA,
      additionalProperties: {
        useSingleRequestParameter: "true",
        groupId: this.pom.groupId,
        artifactId: this.pom.artifactId,
        artifactVersion: this.pom.version,
        invokerPackage: this.packageName,
        apiPackage: `${this.packageName}.api`,
        modelPackage: `${this.packageName}.model`,
        hideGenerationTimestamp: "true",
        additionalModelTypeAnnotations: [
          "@lombok.AllArgsConstructor",
          // Regular lombok builder is not used since an abstract base schema class is also annotated
          "@lombok.experimental.SuperBuilder",
        ].join("\\ "),
      },
      srcDir: path.join("src", "main", "java", ...this.packageName.split(".")),
      normalizers: {
        KEEP_ONLY_FIRST_TAG_IN_OPERATION: true,
      },
      // Do not generate map/list types. Generator will use built in HashMap, ArrayList instead
      generateAliasAsModel: false,
    });
  };
}
