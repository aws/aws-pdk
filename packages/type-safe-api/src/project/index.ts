/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
export * from "./model/type-safe-api-model-build";
export * from "./model/type-safe-api-async-model-build";
export * from "./model/smithy/smithy-project-definition";
export * from "./model/openapi/open-api-project-definition";
export * from "./type-safe-api-project";
export * from "./model/type-safe-api-model-project";
export * from "./model/smithy/smithy-definition";
export * from "./model/openapi/open-api-definition";
export * from "./type-safe-websocket-api-project";
export * from "./model/type-safe-websocket-api-model-project";
export * from "./model/smithy/smithy-async-definition";
export * from "./model/openapi/open-api-async-definition";
export * from "./model/smithy/types";
export * from "./model/openapi/open-api-model-project";
export * from "./model/openapi/open-api-async-model-project";
export * from "./model/smithy/smithy-async-model-project";
export * from "./model/smithy/smithy-model-project";
export * from "./languages";
export * from "./types";
export * from "./typescript-project-options";
export * from "./python-project-options";
export * from "./java-project-options";
