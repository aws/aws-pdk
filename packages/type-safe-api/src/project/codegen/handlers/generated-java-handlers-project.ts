/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import * as path from "path";
import { DependencyType, SampleDir } from "projen";
import { JavaProject } from "projen/lib/java";
import {
  CodeGenerationSourceOptions,
  GeneratedJavaHandlersOptions,
} from "../../types";
import { OpenApiGeneratorHandlebarsIgnoreFile } from "../components/open-api-generator-handlebars-ignore-file";
import { OpenApiGeneratorIgnoreFile } from "../components/open-api-generator-ignore-file";
import { OpenApiToolsJsonFile } from "../components/open-api-tools-json-file";
import { TypeSafeApiCommandEnvironment } from "../components/type-safe-api-command-environment";
import {
  buildCleanOpenApiGeneratedCodeCommand,
  buildInvokeOpenApiGeneratorCommandArgs,
  buildTypeSafeApiExecCommand,
  OtherGenerators,
  TypeSafeApiScript,
} from "../components/utils";
import { GeneratedJavaRuntimeProject } from "../runtime/generated-java-runtime-project";

export interface GeneratedJavaHandlersProjectOptions
  extends GeneratedJavaHandlersOptions,
    CodeGenerationSourceOptions {
  /**
   * The generated java types
   */
  readonly generatedJavaTypes: GeneratedJavaRuntimeProject;
}

export class GeneratedJavaHandlersProject extends JavaProject {
  /**
   * Options configured for the project
   * @private
   */
  private readonly options: GeneratedJavaHandlersProjectOptions;

  /**
   * Source directory
   * @private
   */
  private readonly srcDir: string;

  /**
   * Test directory
   * @private
   */
  private readonly tstDir: string;

  /**
   * Java package name
   */
  public readonly packageName: string;

  constructor(options: GeneratedJavaHandlersProjectOptions) {
    super({
      sample: false,
      junit: false,
      ...(options as any),
    });
    TypeSafeApiCommandEnvironment.ensure(this);
    this.options = options;
    this.packageName = `${this.pom.groupId}.${this.name}.handlers`;
    this.srcDir = path.join(
      "src",
      "main",
      "java",
      ...this.packageName.split(".")
    );
    this.tstDir = path.join(
      "src",
      "test",
      "java",
      ...this.packageName.split(".")
    );

    [
      `${options.generatedJavaTypes.pom.groupId}/${options.generatedJavaTypes.pom.artifactId}@${options.generatedJavaTypes.pom.version}`,
    ]
      .filter(
        (dep) =>
          !this.deps.tryGetDependency(dep.split("@")[0], DependencyType.RUNTIME)
      )
      .forEach((dep) => this.addDependency(dep));

    // Dependency on junit for tests
    this.deps.addDependency(
      "org.junit.jupiter/junit-jupiter-engine@^5.10.0",
      DependencyType.TEST
    );

    // Remove the projen test dependency since otherwise it takes precedence, causing projen to be unavailable at synth time
    this.deps.removeDependency("io.github.cdklabs/projen", DependencyType.TEST);

    // Add a dependency on the generated java types repository
    this.pom.addRepository({
      url: `file://${path.relative(
        this.outdir,
        options.generatedJavaTypes.outdir
      )}/dist/java`,
      id: `${options.generatedJavaTypes.pom.groupId}-${options.generatedJavaTypes.pom.artifactId}-repo`,
    });

    // Ignore everything for the first mustache pass
    const openapiGeneratorIgnore = new OpenApiGeneratorIgnoreFile(this);
    openapiGeneratorIgnore.addPatterns("/*", "**/*", "*");
    // Ignore everything but the handler files for the handlebars pass
    const openapiGeneratorHandlebarsIgnore =
      new OpenApiGeneratorHandlebarsIgnoreFile(this);
    openapiGeneratorHandlebarsIgnore.addPatterns(
      "/*",
      "**/*",
      "*",
      // This will be split into a file per targeted handler
      `!${this.srcDir}/__all_handlers.java`,
      `!${this.tstDir}/__all_tests.java`
    );

    // Add OpenAPI Generator cli configuration
    OpenApiToolsJsonFile.ensure(this).addOpenApiGeneratorCliConfig(
      options.openApiGeneratorCliConfig
    );

    const generateTask = this.addTask("generate");
    generateTask.exec(buildCleanOpenApiGeneratedCodeCommand());
    generateTask.exec(
      buildTypeSafeApiExecCommand(
        TypeSafeApiScript.GENERATE,
        this.buildGenerateCommandArgs()
      )
    );

    this.preCompileTask.spawn(generateTask);

    // Ignore the openapi generator metadata files
    this.gitignore.addPatterns(".openapi-generator");

    // Use the maven shade plugin to build a "super jar" which we can deploy to AWS Lambda
    this.pom.addPlugin("org.apache.maven.plugins/maven-shade-plugin@3.3.0", {
      configuration: {
        createDependencyReducedPom: false,
        transformers: [
          {
            // Transformer required for merging log4j2 plugins cache file
            // https://docs.aws.amazon.com/lambda/latest/dg/java-logging.html#java-logging-cdk
            transformer: {
              "@implementation":
                "com.github.edwgiz.maven_shade_plugin.log4j2_cache_transformer.PluginsCacheFileTransformer",
            },
          },
        ],
      },
      executions: [
        {
          id: "shade-task",
          phase: "package",
          goals: ["shade"],
        },
      ],
      dependencies: [
        "com.github.edwgiz/maven-shade-plugin.log4j2-cachefile-transformer@2.15.0",
      ],
    });

    // Use the maven surefire plugin to run junit tests
    this.pom.addPlugin("org.apache.maven.plugins/maven-surefire-plugin@3.2.1");

    // Log4j2 configuration for powertools logger
    new SampleDir(this, "src/main/resources", {
      files: {
        "log4j2.xml": `<?xml version="1.0" encoding="UTF-8"?>
<Configuration>
    <Appenders>
        <Console name="JsonAppender" target="SYSTEM_OUT">
            <JsonTemplateLayout eventTemplateUri="classpath:LambdaJsonLayout.json" />
        </Console>
    </Appenders>
    <Loggers>
        <Logger name="JsonLogger" level="INFO" additivity="false">
            <AppenderRef ref="JsonAppender"/>
        </Logger>
        <Root level="info">
            <AppenderRef ref="JsonAppender"/>
        </Root>
    </Loggers>
</Configuration>`,
      },
    });
  }

  public buildGenerateCommandArgs = () => {
    return buildInvokeOpenApiGeneratorCommandArgs({
      generator: "java",
      specPath: this.options.specPath,
      generatorDirectory: OtherGenerators.JAVA_LAMBDA_HANDLERS,
      srcDir: this.srcDir,
      tstDir: this.tstDir,
      normalizers: {
        KEEP_ONLY_FIRST_TAG_IN_OPERATION: true,
      },
      extraVendorExtensions: {
        "x-handlers-package": this.packageName,
        "x-runtime-package": this.options.generatedJavaTypes.packageName,
      },
      // Do not generate map/list types. Generator will use built in HashMap, ArrayList instead
      generateAliasAsModel: false,
    });
  };
}
