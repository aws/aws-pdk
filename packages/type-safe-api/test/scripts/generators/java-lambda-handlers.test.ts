/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import os from "os";
import * as path from "path";
import { exec } from "projen/lib/util";
import { GeneratedJavaHandlersProject } from "../../../src/project/codegen/handlers/generated-java-handlers-project";
import { GeneratedJavaRuntimeProject } from "../../../src/project/codegen/runtime/generated-java-runtime-project";
import { withTmpDirSnapshot } from "../../project/snapshot-utils";

describe("Java Handlers Code Generation Script Unit Tests", () => {
  it.each(["handlers.yaml"])("Generates With %s", (spec) => {
    const specPath = path.resolve(__dirname, `../../resources/specs/${spec}`);

    const snapshot = withTmpDirSnapshot(
      os.tmpdir(),
      (outdir) => {
        exec(`cp ${specPath} ${outdir}/spec.yaml`, {
          cwd: path.resolve(__dirname),
        });
        const runtimeOutdir = path.join(outdir, "runtime");
        const runtime = new GeneratedJavaRuntimeProject({
          name: "test-runtime",
          artifactId: "com.aws.pdk.test.runtime",
          groupId: "test",
          version: "1.0.0",
          outdir: runtimeOutdir,
          specPath: "../spec.yaml",
        });
        const handlersOutdir = path.join(outdir, "handlers");
        const project = new GeneratedJavaHandlersProject({
          name: "test-handlers",
          artifactId: "com.aws.pdk.test.handlers",
          groupId: "test",
          version: "1.0.0",
          outdir: handlersOutdir,
          specPath: "../spec.yaml",
          generatedJavaTypes: runtime,
        });
        project.synth();
        exec(
          project.tasks.tryFind("create-openapitools.json")!.steps[0].exec!,
          { cwd: handlersOutdir }
        );
        exec(
          `${path.resolve(
            __dirname,
            "../../../scripts/type-safe-api/generators/generate"
          )} ${project.buildGenerateCommandArgs()}`,
          {
            cwd: handlersOutdir,
          }
        );
      },
      {
        excludeGlobs: GeneratedJavaRuntimeProject.openApiIgnorePatterns,
        parseJson: false,
      }
    );

    expect(
      snapshot[
        "handlers/src/main/java/test/test-handlers/handlers/JavaOneHandler.java"
      ]
    ).toMatchSnapshot();
    expect(
      snapshot[
        "handlers/src/main/java/test/test-handlers/handlers/JavaTwoHandler.java"
      ]
    ).toMatchSnapshot();
    expect(
      snapshot[
        "handlers/src/test/java/test/test-handlers/handlers/JavaOneHandlerTest.java"
      ]
    ).toMatchSnapshot();
    expect(
      snapshot[
        "handlers/src/test/java/test/test-handlers/handlers/JavaTwoHandlerTest.java"
      ]
    ).toMatchSnapshot();

    // Other language handlers should be skipped
    expect(
      Object.keys(snapshot).filter((p) =>
        p.toLowerCase().includes("typescript")
      )
    ).toHaveLength(0);
    expect(
      Object.keys(snapshot).filter((p) => p.toLowerCase().includes("python"))
    ).toHaveLength(0);

    // Split file should be deleted
    expect(
      Object.keys(snapshot).filter((p) =>
        p.toLowerCase().includes("__all_handlers")
      )
    ).toHaveLength(0);
  });
});
