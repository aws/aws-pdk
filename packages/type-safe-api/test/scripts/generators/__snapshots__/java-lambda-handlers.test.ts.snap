// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Java Handlers Code Generation Script Unit Tests Generates With handlers.yaml 1`] = `
"package test.test-handlers.handlers;

import test.test-runtime.runtime.api.interceptors.DefaultInterceptors;
import test.test-runtime.runtime.api.handlers.Interceptor;
import test.test-runtime.runtime.api.handlers.java_one.JavaOne;
import test.test-runtime.runtime.api.handlers.java_one.JavaOneInput;
import test.test-runtime.runtime.api.handlers.java_one.JavaOne500Response;
import test.test-runtime.runtime.api.handlers.java_one.JavaOneRequestInput;
import test.test-runtime.runtime.api.handlers.java_one.JavaOneResponse;
import test.test-runtime.runtime.model.*;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.util.List;

/**
 * Entry point for the AWS Lambda handler for the JavaOne operation.
 * The JavaOne class manages marshalling inputs and outputs.
 */
public class JavaOneHandler extends JavaOne {
    /**
     * Interceptors are initialised once during the lambda "init" phase
     */
    private final List<Interceptor<JavaOneInput>> interceptors = DefaultInterceptors.all();

    /**
     * Use the logger to log requests. The LoggingInterceptor sets up all loggers to include lambda context values in your logs.
     */
    private final Logger log = LogManager.getLogger(JavaOneHandler.class);

    /**
     * Return the interceptors for this handler.
     * You can also use the @Interceptors annotation on the class to add interceptors
     */
    @Override
    public List<Interceptor<JavaOneInput>> getInterceptors() {
        return this.interceptors;
    }

    /**
     * This method is executed prior to the Java SnapStart snapshot being taken.
     * Perform any warmup activities to "prime" your function here. Code in this function will be just-in-time compiled,
     * before the snapshot is taken, and thus optimised ready for the first invocation.
     * For example if your function interacts with DynamoDB, perform a simple read operation here.
     * @see https://aws.amazon.com/blogs/compute/reducing-java-cold-starts-on-aws-lambda-functions-with-snapstart/
     */
    @Override
    public void warmUp() {

    }

    /**
     * Type-safe handler for the JavaOne operation
     */
    @Override
    public JavaOneResponse handle(final JavaOneRequestInput request) {
        log.info("Start JavaOne Operation");

        // TODO: Implement JavaOne Operation. \`input\` contains the request input.
        JavaOneInput input = request.getInput();

        return JavaOne500Response.of(InternalFailureErrorResponseContent.builder()
                .message("Not Implemented!")
                .build());
    }
}

"
`;

exports[`Java Handlers Code Generation Script Unit Tests Generates With handlers.yaml 2`] = `
"package test.test-handlers.handlers;

import test.test-runtime.runtime.api.interceptors.DefaultInterceptors;
import test.test-runtime.runtime.api.handlers.Interceptor;
import test.test-runtime.runtime.api.handlers.java_two.JavaTwo;
import test.test-runtime.runtime.api.handlers.java_two.JavaTwoInput;
import test.test-runtime.runtime.api.handlers.java_two.JavaTwo500Response;
import test.test-runtime.runtime.api.handlers.java_two.JavaTwoRequestInput;
import test.test-runtime.runtime.api.handlers.java_two.JavaTwoResponse;
import test.test-runtime.runtime.model.*;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.util.List;

/**
 * Entry point for the AWS Lambda handler for the JavaTwo operation.
 * The JavaTwo class manages marshalling inputs and outputs.
 */
public class JavaTwoHandler extends JavaTwo {
    /**
     * Interceptors are initialised once during the lambda "init" phase
     */
    private final List<Interceptor<JavaTwoInput>> interceptors = DefaultInterceptors.all();

    /**
     * Use the logger to log requests. The LoggingInterceptor sets up all loggers to include lambda context values in your logs.
     */
    private final Logger log = LogManager.getLogger(JavaTwoHandler.class);

    /**
     * Return the interceptors for this handler.
     * You can also use the @Interceptors annotation on the class to add interceptors
     */
    @Override
    public List<Interceptor<JavaTwoInput>> getInterceptors() {
        return this.interceptors;
    }

    /**
     * This method is executed prior to the Java SnapStart snapshot being taken.
     * Perform any warmup activities to "prime" your function here. Code in this function will be just-in-time compiled,
     * before the snapshot is taken, and thus optimised ready for the first invocation.
     * For example if your function interacts with DynamoDB, perform a simple read operation here.
     * @see https://aws.amazon.com/blogs/compute/reducing-java-cold-starts-on-aws-lambda-functions-with-snapstart/
     */
    @Override
    public void warmUp() {

    }

    /**
     * Type-safe handler for the JavaTwo operation
     */
    @Override
    public JavaTwoResponse handle(final JavaTwoRequestInput request) {
        log.info("Start JavaTwo Operation");

        // TODO: Implement JavaTwo Operation. \`input\` contains the request input.
        JavaTwoInput input = request.getInput();

        return JavaTwo500Response.of(InternalFailureErrorResponseContent.builder()
                .message("Not Implemented!")
                .build());
    }
}



"
`;

exports[`Java Handlers Code Generation Script Unit Tests Generates With handlers.yaml 3`] = `
"package test.test-handlers.handlers;

import test.test-runtime.runtime.api.handlers.java_one.JavaOneInput;
import test.test-runtime.runtime.api.handlers.java_one.JavaOne500Response;
import test.test-runtime.runtime.api.handlers.java_one.JavaOneRequestInput;
import test.test-runtime.runtime.api.handlers.java_one.JavaOneRequestParameters;
import test.test-runtime.runtime.api.handlers.java_one.JavaOneResponse;
import org.junit.jupiter.api.Test;

import java.util.HashMap;

import static org.junit.jupiter.api.Assertions.assertEquals;

/**
 * Tests for JavaOneHandler
 */
public class JavaOneHandlerTest {
    @Test
    public void shouldReturnNotImplementedError() {
        // TODO: Update the test as appropriate when you implement your handler
        JavaOneResponse response = new JavaOneHandler().handle(JavaOneRequestInput.builder()
                .interceptorContext(new HashMap<>())
                .input(JavaOneInput.builder()
                        .requestParameters(JavaOneRequestParameters.builder()
                                // Add request parameters here...
                                .build())
                        // If the request has a body you can add it here...
                        .build())
                .build());
        assertEquals(500, response.getStatusCode());

        // Cast to the particular type of response returned to access the typed response body
        assertEquals("Not Implemented!", ((JavaOne500Response) response).getTypedBody().getMessage());
    }
}

"
`;

exports[`Java Handlers Code Generation Script Unit Tests Generates With handlers.yaml 4`] = `
"package test.test-handlers.handlers;

import test.test-runtime.runtime.api.handlers.java_two.JavaTwoInput;
import test.test-runtime.runtime.api.handlers.java_two.JavaTwo500Response;
import test.test-runtime.runtime.api.handlers.java_two.JavaTwoRequestInput;
import test.test-runtime.runtime.api.handlers.java_two.JavaTwoRequestParameters;
import test.test-runtime.runtime.api.handlers.java_two.JavaTwoResponse;
import org.junit.jupiter.api.Test;

import java.util.HashMap;

import static org.junit.jupiter.api.Assertions.assertEquals;

/**
 * Tests for JavaTwoHandler
 */
public class JavaTwoHandlerTest {
    @Test
    public void shouldReturnNotImplementedError() {
        // TODO: Update the test as appropriate when you implement your handler
        JavaTwoResponse response = new JavaTwoHandler().handle(JavaTwoRequestInput.builder()
                .interceptorContext(new HashMap<>())
                .input(JavaTwoInput.builder()
                        .requestParameters(JavaTwoRequestParameters.builder()
                                // Add request parameters here...
                                .build())
                        // If the request has a body you can add it here...
                        .build())
                .build());
        assertEquals(500, response.getStatusCode());

        // Cast to the particular type of response returned to access the typed response body
        assertEquals("Not Implemented!", ((JavaTwo500Response) response).getTypedBody().getMessage());
    }
}



"
`;
